### PR TITLE
[ADH-4549] Implement endpoint for retrieving information about the logged-in user

### PR DESCRIPTION
**Pull request by tigrulya-exe**
- Added endpoint for retrieving information about the logged-in user
- Replaced `null` user with `anonymous` one in case of disabled authentication
- Added `smart.rest.server.auth.spnego.keytab` config to optionally provide SSM server keytab for `HTTP` principal separately from SSM service keytab (`smart.server.keytab.file`)

### CODE REVIEW

File: smart-web-server/src/main/java/org/smartdata/server/generated/api/SystemApi.java [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+        }
+    )
+    @RequestMapping(
+        method = RequestMethod.GET,
+        value = "/api/v2/system/current-user",
+        produces = { "application/json" }
+    )
+    @ResponseStatus(HttpStatus.OK)
+    
+    default UserInfoDto getCurrentUser(
```
Comment (VitekArkhipov) [inline smart-web-server/src/main/java/org/smartdata/server/generated/api/SystemApi.java:76]: please, reformat the code

File: smart-web-server/src/main/java/org/smartdata/server/generated/api/SystemApiController.java [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+import javax.annotation.Generated;
+
+import java.util.Optional;
+
+@Generated(value = "org.openapitools.codegen.languages.SpringCodegen")
+@RestController
+@RequestMapping("${openapi.sSMAPIDocumentation.base-path:}")
```
Comment (VitekArkhipov) [inline smart-web-server/src/main/java/org/smartdata/server/generated/api/SystemApiController.java:30]: may be here we should have smth like: ${openapi.sSMAPIDocumentation.base-path:} + /api/v2/system and in SystemApi interface on the method getCurrentUser() smth like this: /current-user ?

File: smart-web-server/src/main/java/org/smartdata/server/generated/api/SystemApi.java (hunk 1 above)
Comment (tigrulya-exe) [reply smart-web-server/src/main/java/org/smartdata/server/generated/api/SystemApi.java:76]: This class is generated by openapi maven plugin

File: smart-web-server/src/main/java/org/smartdata/server/generated/api/SystemApiController.java (hunk 2 above)
Comment (tigrulya-exe) [reply smart-web-server/src/main/java/org/smartdata/server/generated/api/SystemApiController.java:30]: Same as above

**Review by iamlapa — APPROVED** [review]

**Pull request merged**